src/Components/ClustersListTable/ClustersListTable.cy.js

```diff
@@ -199,6 +201,65 @@ describe('data', () => {
   });
 });
 
+const urlParamsList = [
+  'text=test&version=4.9.0&hits=4',
+  'text=test&version=4.9.0&hits=3',
+  'text=test&version=4.10.0&hits=2',
+  'text=test&hits=1&version=4.2.35',
+];
+
+urlParamsList.forEach((urlParams, index) => {
+  describe(`pre-filled url search parameters ${index}`, () => {
+    beforeEach(() => {
+      mount(
+        <MemoryRouter
+          initialEntries={[`/clusters?${urlParams}`]}
+          initialIndex={0}
+        >
+          <Intl>
+            <Provider store={getStore()}>
+              <ClustersListTable
+                query={{
+                  isError: false,
+                  isFetching: false,
+                  isUninitialized: false,
+                  isSuccess: true,
+                  data: clusters,
+                  refetch: cy.stub(),
+                }}
+              />
+            </Provider>
+          </Intl>
+        </MemoryRouter>
+      );
+    });
+
+    it('recognizes all parameters', () => {
+      const urlSearchParameters = new URLSearchParams(urlParams);
+      for (const [key, value] of urlSearchParameters) {
+        if (key == 'text') {
+          hasChip('Name', value, 'cluster');
+          cy.get('.pf-m-fill > .pf-c-form-control').should('have.value', value);
+        } else {
+          value.split(',').forEach((it) => {
+            const [group, item] = urlParamConvert(
+              'hits',
```

Comment:
Definitely, this is not something you want to do. Basically you are setting all items in the else clause to check for `hits`. That shouldn't be the case.

Reply:
I added an array 
`const clusterListFilterValues = ['hits'];`
to store values and utilized _.find() lodash method to search the values in the array